src/engine.cc

```diff
@@ -236,9 +237,15 @@ void EngineController::SetCacheSize(int size) { cache_.SetCapacity(size); }
 void EngineController::EnsureReady() {
   UpdateTBAndNetwork();
   std::unique_lock<RpSharedMutex> lock(busy_mutex_);
+  // If a UCI host is waiting for our ready response, we can consider the move
```

Comment:
That's not what arena does, at least.
Before we had a separate thread for tree GC, `position startpos moves ...` started to free unused parts of a tree, and we didn't reply `readyok` until it's done, but still timed out a lot because clock ticked.

Reply:
arena doesn't even call isready after set position at all.
cutechess does and doesn't start the timer until readyok.